src/app/helpers/decrypt.js

```diff
@@ -73,7 +73,14 @@ const ACTIONS = {
 export const prepareContact = async (contact, { publicKeys, privateKeys }) => {
     const { Cards } = contact;
 
-    const data = await Promise.all(Cards.map((card) => ACTIONS[card.Type](card, { publicKeys, privateKeys })));
+    const data = await Promise.all(
+        Cards.map(async (card) => {
+            if (!ACTIONS[card.Type]) {
+                return { error: FAIL_TO_READ };
```

Comment:
In a list of promise, better a rejected promise than an object with an error property.

Reply:
Here we need the promise to continue decrypting cards even if there are errors for some of them. That's why we return an object ;)